File: front/app/models/weather-location.js
import Model, { attr, hasMany } from '@ember-data/model';

export default class WeatherLocationModel extends Model {
  // --------------------------------------------------
  // Relationships
  // --------------------------------------------------
  @hasMany('user', {
    async: true,
    inverse: 'weatherLocations',
  })
  users;

  @hasMany('weather-forecast', {
    async: true,
    inverse: 'weatherLocation',
  })
  weatherForecasts;

  // --------------------------------------------------
  // Attributes
  // --------------------------------------------------
  @attr('string') name;
  @attr('string') region;
  @attr('string') country;
  @attr('number') lat;
  @attr('number') lon;
  // @attr('moment-date') createdAt;

  get currentForecast() {
    console.log("Current fc: ", this.weatherForecasts);
    return this.weatherForecasts.firstObject;
  }
}

File: api/app/models/weather_location.rb
# == Schema Information
#
# Table name: weather_locations
#
#  id         :bigint           not null, primary key
#  country    :string
#  lat        :decimal(, )
#  lon        :decimal(, )
#  name       :string
#  region     :string
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
# Indexes
#
#  index_lat_lon                    (lat,lon)
#  index_weather_locations_on_name  (name)
#
class WeatherLocation < ApplicationRecord
  # ----------------------------------------
  # RELATIONSHIPS
  # ----------------------------------------
  has_many :weather_forecasts, dependent: :destroy
  has_and_belongs_to_many :users

  # ----------------------------------------
  # SETUP
  # ----------------------------------------
  validates :name, presence: true
  validates :country, presence: true
  validates :lat, presence: true
  validates :lon, presence: true
end
